mesonbuild/interpreter/type_checking.py

```diff
@@ -841,10 +840,16 @@ def _convert_darwin_versions(val: T.List[T.Union[str, int]]) -> T.Optional[T.Tup
     )
 ]
 
+def _pkgconfig_define_convertor(x: T.List[str]) -> PkgConfigDefineType:
+    if x:
+        keys = itertools.islice(x, 0, None, 2)
+        vals = itertools.islice(x, 1, None, 2)
+        return tuple(zip(keys, vals))
```

Comment:
nitpicking: This can be a one line: `zip(x[0::2], x[1::2])`. Even works on empty list so `if x:` is not needed.

OOC is there a reason to prefer `tuple[tuple]` rather than `list[tuple]`?

Reply:
I already do return None. As for the Tuple, the result is used as a cache key and must be hashable, which is also why the original code used tuples too.

As far as making it one line goes, I think I locked myself into a mental box by originally trying to find an itertools solution, discovering that while one does exist it's specific to python 3.12, and then having itertools on my mind when trying to roll my own from scratch.

Reply:
> I already do return None.

Oops I'm blind, you're right.